src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java

```diff
@@ -92,7 +106,54 @@
  * @author Kohsuke Kawaguchi
  */
 public class ActiveDirectorySecurityRealm extends AbstractPasswordBasedSecurityRealm {
+  static public class SSOOptions {
+    public final String userField;
+    public final String defaultGroup;
+    public final String checkMode;
+    public final String jenkinsUsernameSuffix;
+    
+    public String getJenkinsUsernameSuffix()
+    {
+      return jenkinsUsernameSuffix;
+    }
+
     /**
+     * Field of the DN to look at.
+     */
+    public String getUserField() {
+        return this.userField;
+    }
+  
+    /**
+     * Field of the Sso Case mode to be converted in another case
+     */
+    public String getCheckMode() {
+        return this.checkMode;
+    }
+    
+    public String getDefaultGroup() {
+      return this.defaultGroup;
+    }
+    
+  
+    @DataBoundConstructor
+    public SSOOptions(final String userField, final String defaultGroup, final String checkMode,final String jenkinsUsernameSuffix){
+      this.userField=userField;
+      this.checkMode=checkMode;
+      this.jenkinsUsernameSuffix=jenkinsUsernameSuffix;
+      this.defaultGroup=defaultGroup;
+    }
+  }
+  
+  public final SSOOptions ssoEnabled;
+  /**
+   * Is SSO Enabled
```

Comment:
IMO this is not readable enough. Maybe that code started with only a boolean and got refactored? I don't see why this variable should be called "ssoEnabled"